src/python/py_image_generation_pipelines.cpp

```diff
@@ -68,7 +53,70 @@ auto text2image_generate_docstring = R"(
     :rtype: ov.Tensor
 )";
 
+// Trampoline class to support inheritance from Generator in Python
+class PyGenerator : public ov::genai::Generator {
+public:
+    float next() override {
+        PYBIND11_OVERRIDE_PURE(float, Generator, next);
+    }
 
+    ov::Tensor randn_tensor(const ov::Shape& shape) override {
+        PYBIND11_OVERRIDE(ov::Tensor, Generator, randn_tensor, shape);
+    }
+};
+
+py::list to_py_list(const ov::Shape shape) {
+    py::list py_shape;
+    for (auto d : shape)
+        py_shape.append(d);
+
+    return py_shape;
+}
+
+class TorchGenerator : public ov::genai::CppStdGenerator {
+    py::module_ m_torch;
+    py::object m_torch_generator, m_float32;
+    py::object m_torch_tensor; // we need to hold torch.Tensor to avoid memory destruction
+public:
+    explicit TorchGenerator(uint32_t seed) : CppStdGenerator(seed) {
+        try {
+            m_torch = py::module_::import("torch");
+            m_torch_generator = m_torch.attr("Generator")("device"_a="cpu").attr("manual_seed")(seed);
+            m_float32 = m_torch.attr("float32");
+        } catch (const py::error_already_set& e) {
+            if (e.matches(PyExc_ModuleNotFoundError)) {
+                PyErr_WarnEx(PyExc_ImportWarning, "'torch' module is not installed. Random generation will fall back to 'CppStdGenerator'", 1);
```

Comment:
Printing anything from a library isn't really welcome (at least for me). It's an application's (sample's in our case) responsibility to communicate with a user.
My proposal is the following. `TorchGenerator` should throw if `torch` is missing. Introduce a function in samples that picks an available generator and prints this warning.

I have another concern. Why do we need a C++ wrapper over python in the first place. Samples themselves could define a thin generator in python similar to this `TorchGenerator` because there's no benefit in jumping between C++ and Python multiple times. That implementation would also serve as an example generator.

Reply:
> My proposal is the following. TorchGenerator should throw if torch is missing

Having slept on it, I agree that throwing exception is better idea.